File: dplyr/01_read.R
# # Read data from a delimited text file into an R data frame

# Use the built-in R function `read.table()` to read data
# from a delimited text file into an R data frame
games <- read.table(
  "data/games/games.csv",
  header = TRUE,
  sep = ","
)

# The default delimiter for `read.table()` is any
# whitespace. Use the `sep` argument to specify a 
# different delimiter. By default, `read.table()`
# expects the file to have no header row. To treat the
# first row of the file as a header row containing the
# column names, specify `header = TRUE`

# For details about the `read.table()` function, run the
# help command:
?read.table

# Alternatively, use the `read.csv()` function which
# expects a comma delimiter and a header row by default
games <- read.csv("data/games/games.csv")

# These functions return a `data.frame` object
class(games)

# View the data
games


# Instead of using R's built-in functions for reading
# data, you can use the functions in the 
# [readr](https://readr.tidyverse.org) package. These
# functions behave slightly differently, and return
# ["tibble"](https://tibble.tidyverse.org) objects, which
# have class `tbl_df`. Tibbles are still data frames, but
# they have some different behaviors.

# Load the readr package
library(readr)

# Call the function `read_csv()` (underscore instead of
# dot) to read data from a comma-delimited file into a 
# tibble.
games <- read_csv("data/games/games.csv")

# These functions return a `tbl_df` object, which also
# inherits class `data.frame`
class(games)

# View the data as a tibble
games

# View the data as a data frame
as.data.frame(games)

File: pandas/01_read.py
# # Read data from a delimited text file into a pandas DataFrame

# Import modules
import numpy as np
import pandas as pd

# Read data using the `read_table` function
games = pd.read_table('data/games/games.csv', sep=',')

# The default delimiter for `read_table` is tab (`'\t'`).
# Use the `sep` parameter to specify a different
# delimiter

# For details about the `read_table` function, run the
# help command:
pd.read_table?

# Alternatively, use the `read_csv` function which uses
# a comma delimiter by default
games = pd.read_csv('data/games/games.csv')

# pandas provides other functions for reading other
# formats of file-based data. See the 
# [Input/Output section of the pandas API reference](https://pandas.pydata.org/pandas-docs/stable/api.html#input-output)
# for details.

# These functions return a `DataFrame` object
type(games)

# View the data
games

# To display the pandas DataFrame in a scrollable
# grid, set the pandas option
# `display.html.table_schema` to `True`:

pd.set_option("display.html.table_schema", True)

# View the data in a scrollable grid
games

# Change to previous setting
pd.set_option("display.html.table_schema", False)
